src/makeCalculator.js

```diff
@@ -4,7 +4,35 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(num) {
+      this.result += num;
+    },
+    subtract(num) {
+      this.result -= num;
+    },
+    multiply(num) {
+      this.result *= num;
+    },
+    divide(num) {
+      if (num === 0) {
+        return null;
+      } else {
+        this.result /= num;
+      }
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(operation, num) {
+      operation.call(this, num);
```

Comment:
The `operate` method assumes that the `operation` parameter is a valid function. It would be beneficial to add a check to ensure that `operation` is indeed a function before calling it. This would prevent runtime errors if an invalid operation is passed.